node-api/handlers/proof/types/response.go

```diff
@@ -69,3 +70,27 @@ type ValidatorWithdrawalCredentialsResponse struct {
 	// block. In the Electra fork, z is 6350779162034177.
 	WithdrawalCredentialsProof []common.Root `json:"withdrawal_credentials_proof"`
 }
+
+// ValidatorBalanceResponse is the response for the
+// `/proof/validator_balance/{timestamp_id}/{validator_index}` endpoint.
+type ValidatorBalanceResponse struct {
+	// BeaconBlockHeader is the block header of which the hash tree root is the
+	// beacon block root to verify against.
+	BeaconBlockHeader *ctypes.BeaconBlockHeader `json:"beacon_block_header"`
+
+	// BeaconBlockRoot is the beacon block root for this slot.
+	BeaconBlockRoot common.Root `json:"beacon_block_root"`
+
+	// ValidatorBalance is the balance of the requested validator.
+	ValidatorBalance math.Gwei `json:"validator_balance"`
+
+	// BalanceLeaf is the leaf containing the validator's balance along with up
+	// to 3 other validators' balances (packed 4 per leaf).
+	BalanceLeaf common.Root `json:"balance_leaf"`
```

Comment:
@bearmigiano important to note for staking pools: The leaf of the proof is not just the HashTreeRoot of the balance. Since the leaf contains multiple balances (not just the one we are proving), you must use this one while verifying the proof.